jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -527,14 +577,81 @@ public CRIUSupport registerPostRestoreHook(Runnable hook) {
 	 *
 	 */
 	public CRIUSupport registerPreCheckpointHook(Runnable hook) {
+		return registerPreCheckpointHook(hook, HookMode.SINGLE_THREAD_MODE, USER_HOOKS_PRIORITY);
+	}
+
+	/**
+	 * User hook that is run before checkpointing the JVM.
+	 *
+	 * If single-threaded mode is requested, no other application threads will be
+	 * active. Users should avoid synchronization of objects that are not owned by
+	 * the thread, terminally blocking operations and launching new threads in the
+	 * hook. If the thread attempts to acquire a lock that it doesn't own, an
+	 * exception will be thrown.
+	 *
+	 * If multi-threaded mode is requested, the hook will be run alongside all other
+	 * active Java threads.
+	 *
+	 * High-priority hooks are run last before checkpoint, and vice-versa for
+	 * low-priority hooks. The priority of the hook is with respect to the other
+	 * hooks run within that mode. Multi-threaded hooks are implicitly lower
+	 * priority than single-threaded hooks. Ie. the lowest priority single-threaded
+	 * hook is a higher priority than the highest priority multi-threaded hook.
```

Comment:
Add that hooks of the same mode with the same priority are run in random order

Reply:
Add the comment